### PR TITLE
Add new unit test coverage

### PR DESCRIPTION
**Pull request by oliveromahony**
### Proposed changes

Tidied up some testing and made the code more readable

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: test/helpers.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	}
+}
+
+func CreateProtoTime(t *testing.T, timeString string) *timestamppb.Timestamp {
+	t.Helper()
+	newTime, err := time.Parse(time.RFC3339, timeString)
+	protoTime := timestamppb.New(newTime)
+	if err != nil {
+		assert.Fail(t, fmt.Sprintf("failed on creating timestamp %s", protoTime))
+	}
```
Comment (dhurley) [inline test/helpers.go:41]:
```suggestion
        require.NoError(t, err)
	protoTime := timestamppb.New(newTime)
```

File: test/helpers.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func CreateFileWithErrorCheck(t *testing.T, dir, fileName string) *os.File {
```
Comment (dhurley) [inline test/helpers.go:19]: The `CreateFileWithErrorCheck` and `RemoveFileWithErrorCheck` functions seem a bit of an overkill to me. Don't think we need to make our own functions for creating files and directories.

**Review by dhurley — COMMENTED** [review]
I see that in some places you have renamed `dataplane` to `dataPlane`. I think you have missed a few other places in the codebase where the name would need to be changed also to be consistent.

File: test/helpers.go (hunk 2 above)
Comment (oliveromahony) [reply test/helpers.go:19]: it simplifies 2 lines of code that appears numerous times in the tests

**Comment by oliveromahony** [comment]
> I see that in some places you have renamed `dataplane` to `dataPlane`. I think you have missed a few other places in the codebase where the name would need to be changed also to be consistent.

I just did it in some places and not all to make the PR smaller. I tried to do it in the files I touched mostly

**Review by aphralG — APPROVED** [review]

File: test/helpers.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	testConf, err := os.CreateTemp(dir, fileName)
+	require.NoError(t, err)
+
+	return testConf
+}
+
+func RemoveFileWithErrorCheck(t *testing.T, fileName string) {
+	t.Helper()
+	err := os.Remove(fileName)
+	if err != nil {
```
Comment (dhurley) [inline test/helpers.go:30]: instead of the if statement would `require.NoError(err)` be better?

File: test/helpers.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	}
+}
+
+func CreateProtoTime(t *testing.T, timeString string) *timestamppb.Timestamp {
+	t.Helper()
+	newTime, err := time.Parse(time.RFC3339, timeString)
+	require.NoError(t, err)
+
+	protoTime := timestamppb.New(newTime)
+	if err != nil {
```
Comment (dhurley) [inline test/helpers.go:41]: You can remove this if statement

File: test/helpers.go (hunk 2 above)
Comment (dhurley) [reply test/helpers.go:19]: I guess this is used a far bit alright in the test code. Just want to be careful that we dont overuse this pattern and end up with a lot of wrapper test functions that only check errors returned from another function.

File: internal/plugin/dataplane_server_test.go [hunk 5]
```diff
@@ -26,39 +26,68 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestDataplaneServer_Init(t *testing.T) {
-	dataplaneServer := NewDataplaneServer(&config.Config{}, slog.Default())
+func TestDataPlaneServer_Init(t *testing.T) {
+	dataPlaneServer := NewDataplaneServer(&config.Config{}, slog.Default())
```
Comment (dhurley) [inline internal/plugin/dataplane_server_test.go:30]: Should `NewDataplaneServer` be `NewDataPlaneServer` now?

File: internal/plugin/dataplane_server.go [hunk 6]
```diff
@@ -88,10 +88,10 @@ func (dps *DataplaneServer) run(_ context.Context) {
 	server.Use(sloggin.NewWithConfig(dps.logger, sloggin.Config{DefaultLevel: slog.LevelDebug}))
```
Comment (dhurley) [inline internal/plugin/dataplane_server.go:88]: Should the file be called `data_plane_server.go` now?

File: test/data.go [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+		FilePath:    path,
+		InstanceId:  instanceID,
+		FileContent: content,
+	}
+}
+
+func CreateProtoTime(timeString string) *timestamppb.Timestamp {
+	newTime, err := time.Parse(time.RFC3339, timeString)
+	if err != nil {
+		slog.Error("failed to parse time")
```
Comment (dhurley) [inline test/data.go:56]: Should we not return an error here?

File: test/helpers.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
+// Copyright (c) F5, Inc.
```
Comment (dhurley) [inline test/helpers.go:1]: We are using `require.NoError`, `t.Fail` and `assert.Fail` in this file when checking errors. We should pick one way and be consistent.

Comment (oliveromahony) [reply test/helpers.go:1]: that was updated based on previous comments

**Review by olli-holmala — APPROVED** [review]

**Review by dhurley — APPROVED** [review]

**Pull request merged**